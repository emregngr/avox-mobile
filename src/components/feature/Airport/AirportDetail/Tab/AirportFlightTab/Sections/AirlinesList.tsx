import React from 'react'

import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { AirlineRowCard } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Cards/AirlineRowCard'

interface AirlinesListProps {
  airlines: string[]
  iconColor: string,
  title: string
}

export const AirlinesList = ({ airlines, iconColor, title }: AirlinesListProps) => (
  <AirportSectionRow title={title}>
    {airlines?.map((airline, index) => (
      <AirlineRowCard airline={airline} iconColor={iconColor} key={index} />
      ))}
  </AirportSectionRow>
  )
