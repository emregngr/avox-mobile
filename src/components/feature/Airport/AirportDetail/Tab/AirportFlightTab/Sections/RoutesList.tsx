import React from 'react'

import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { RouteRowCard } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Cards/RouteRowCard'
import type { AirportRoute } from '@/types/feature/airport'

interface RoutesListProps {
  iconColor: string
  routes: AirportRoute[]
  title: string
}

export const RoutesList = ({ iconColor, routes, title }: RoutesListProps) => (
  <AirportSectionRow title={title}>
    {routes?.map((route, index) => (
      <RouteRowCard
        destinationIata={route.destinationIata}
        frequency={route.frequency}
        iconColor={iconColor}
        key={index}
      />
    ))}
  </AirportSectionRow>
)
