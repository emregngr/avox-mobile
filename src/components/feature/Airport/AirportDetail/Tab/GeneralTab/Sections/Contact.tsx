import { router } from 'expo-router'
import React, { useCallback, useMemo } from 'react'
import { Linking } from 'react-native'

import { AirportRowItem } from '@/components/feature/Airport/AirportDetail/AirportRowItem'
import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'

interface ContactProps {
  airportInfo: Airport['airportInfo']
}

export const Contact = ({ airportInfo }: ContactProps) => {
  const { selectedLocale } = useLocaleStore()

  const { contactInfo: { email, phone } = {}, website } = airportInfo ?? {}

  const handlePhonePress = useCallback(async () => {
    if (phone) {
      await Linking?.openURL(`tel:${phone}`)
    }
  }, [phone])

  const handleEmailPress = useCallback(async () => {
    if (email) {
      await Linking?.openURL(`mailto:${email}`)
    }
  }, [email])

  const handleWebsitePress = useCallback(() => {
    if (website) {
      const formattedUrl = website?.startsWith('http') ? website : `https://${website}`

      router.navigate({
        params: {
          title: website,
          webViewUrl: formattedUrl,
        },
        pathname: '/web-view-modal',
      })
    }
  }, [website])

  const localeStrings = useMemo(
    () => ({
      contactInformation: getLocale('contactInformation'),
      email: getLocale('email'),
      phone: getLocale('phone'),
      website: getLocale('website'),
    }),
    [selectedLocale],
  )

  return (
    <AirportSectionRow title={localeStrings.contactInformation}>
      <AirportRowItem
        icon="globe"
        label={localeStrings.website}
        onPress={handleWebsitePress}
        value={website}
      />

      <AirportRowItem
        icon="call"
        label={localeStrings.phone}
        onPress={handlePhonePress}
        value={phone}
      />

      <AirportRowItem
        icon="mail"
        label={localeStrings.email}
        onPress={handleEmailPress}
        value={email}
      />
    </AirportSectionRow>
  )
}
