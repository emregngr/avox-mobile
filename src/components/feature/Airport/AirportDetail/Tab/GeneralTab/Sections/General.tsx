import React, { useMemo } from 'react'

import { AirportRowItem } from '@/components/feature/Airport/AirportDetail/AirportRowItem'
import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'
import { formatNumber } from '@/utils/feature/formatNumber'

interface GeneralProps {
  airportInfo: Airport['airportInfo']
  operations: Airport['operations']
}

export const General = ({ airportInfo, operations }: GeneralProps) => {
  const { selectedLocale } = useLocaleStore()

  const { employeeCount, foundingYear } = airportInfo ?? {}
  const { airportType, is24Hour, scheduledService } = operations ?? {}

  const yearsSinceEstablishment = useMemo(
    () => (foundingYear ? new Date().getFullYear() - parseInt(foundingYear) : 0),
    [foundingYear],
  )

  const localeStrings = useMemo(
    () => ({
      airportInfo: getLocale('airportInfo'),
      airportType: getLocale('airportType'),
      numberOfEmployees: getLocale('numberOfEmployees'),
      open24Hours: getLocale('open24Hours'),
      scheduledService: getLocale('scheduledService'),
      year: getLocale('year'),
      yearOfEstablishment: getLocale('yearOfEstablishment'),
    }),
    [selectedLocale],
  )

  const formattedYearOfEstablishment = useMemo(
    () => `${foundingYear} (${yearsSinceEstablishment} ${localeStrings.year})`,
    [foundingYear, yearsSinceEstablishment, localeStrings],
  )

  const formattedEmployeeCount = useMemo(() => formatNumber(employeeCount), [employeeCount])

  const airportTypeValue = useMemo((): string => {
    if (airportType === 'small_airport') {
      return getLocale('small')
    } else if (airportType === 'medium_airport') {
      return getLocale('medium')
    } else if (airportType === 'large_airport') {
      return getLocale('large')
    } else if (airportType === 'mega_airport') {
      return getLocale('mega')
    }

    return airportType?.replace('_', ' ')
  }, [airportType])

  const is24HourValue = useMemo(() => (is24Hour ? getLocale('yes') : getLocale('no')), [is24Hour])

  const scheduledServiceValue = useMemo(
    () => (scheduledService ? getLocale('yes') : getLocale('no')),
    [scheduledService],
  )

  return (
    <AirportSectionRow title={localeStrings.airportInfo}>
      <AirportRowItem
        icon="calendar"
        label={localeStrings.yearOfEstablishment}
        value={formattedYearOfEstablishment}
      />

      <AirportRowItem
        icon="people"
        label={localeStrings.numberOfEmployees}
        value={formattedEmployeeCount}
      />

      <AirportRowItem icon="airplane" label={localeStrings.airportType} value={airportTypeValue} />

      <AirportRowItem icon="time" label={localeStrings.open24Hours} value={is24HourValue} />

      <AirportRowItem
        icon="calendar-outline"
        label={localeStrings.scheduledService}
        value={scheduledServiceValue}
      />
    </AirportSectionRow>
  )
}
