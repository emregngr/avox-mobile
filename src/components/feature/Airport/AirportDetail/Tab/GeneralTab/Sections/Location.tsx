import React, { useMemo } from 'react'

import { AirportRowItem } from '@/components/feature/Airport/AirportDetail/AirportRowItem'
import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'
import { formatNumber } from '@/utils/feature/formatNumber'

interface LocationProps {
  operations: Airport['operations']
}

export const Location = ({ operations }: LocationProps) => {
  const { selectedLocale } = useLocaleStore()

  const {
    country,
    location: { city, elevationFt },
  } = operations || {}

  const localeStrings = useMemo(
    () => ({
      city: getLocale('city'),
      country: getLocale('country'),
      elevation: getLocale('elevation'),
      locationInformation: getLocale('locationInformation'),
    }),
    [selectedLocale],
  )

  return (
    <AirportSectionRow title={localeStrings.locationInformation}>
      <AirportRowItem icon="business" label={localeStrings.city} value={city} />

      <AirportRowItem icon="earth" label={localeStrings.country} value={country} />

      <AirportRowItem
        icon="trending-up"
        label={localeStrings.elevation}
        value={formatNumber(elevationFt)}
      />
    </AirportSectionRow>
  )
}
