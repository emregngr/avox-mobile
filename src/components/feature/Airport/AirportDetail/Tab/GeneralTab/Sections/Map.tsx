import { AppleMaps } from 'expo-maps'
import type { AppleMapsMarker } from 'expo-maps/build/apple/AppleMaps.types'
import React, { useCallback, useMemo } from 'react'
import { Platform, View } from 'react-native'
import { WebView } from 'react-native-webview'

import { ThemedText } from '@/components/common/ThemedText'
import { useMapActions } from '@/hooks/maps/useMapAction'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airport } from '@/types/feature/airport'
import type { RegionKey } from '@/types/feature/region'

const zoomLevel = 12

const createDynamicGoogleMapsHTML = (
  name: string,
  latitude: number,
  longitude: number,
  selectedTheme: 'light' | 'dark',
) => {
  const embedUrl = `https://maps.google.com/maps?q=${latitude},${longitude}&z=${zoomLevel}&output=embed`

  return `
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <title>${name} HaritasÄ±</title>
      <style>
        body {
          margin: 0;
          padding: 0;
          font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
          ${selectedTheme === 'dark' ? 'background-color: #1a1a1a;' : 'background-color: #ffffff;'}
        }

        .map-container {
          width: 100%;
          height: 250px;
          position: relative;
          overflow: hidden;
          ${
            selectedTheme === 'dark'
              ? 'border: 1px solid #333; border-radius: 8px;'
              : 'border: 1px solid #e0e0e0; border-radius: 8px;'
          }
        }

        iframe {
          width: 100%;
          height: 260px;
          border: none;
          position: absolute;
          top: 0;
          left: 0;
          ${
            selectedTheme === 'dark'
              ? 'filter: invert(90%) hue-rotate(180deg) brightness(95%) contrast(90%);'
              : ''
          }
        }
      </style>
    </head>
    <body>
      <div class="map-container">
        <iframe
          src="${embedUrl}"
          allowfullscreen=""
          loading="lazy"
          referrerpolicy="no-referrer-when-downgrade">
        </iframe>
      </div>
    </body>
    </html>
  `
}

interface MapProps {
  airportData: Airport
}

export const Map = ({ airportData }: MapProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()
  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { iataCode, id, name, operations } = airportData ?? {}
  const {
    location: {
      coordinates: { latitude, longitude },
    },
    region,
  } = operations ?? {}

  const googleMapsHTML = useMemo(
    () => createDynamicGoogleMapsHTML(name, latitude, longitude, selectedTheme),
    [name, latitude, longitude, selectedTheme],
  )

  const markerColor = useMemo(() => colors?.[region?.toLowerCase() as RegionKey], [colors, region])

  const appleMarkers: AppleMapsMarker[] = useMemo(
    () => [
      {
        coordinates: { latitude, longitude },
        id: name,
        systemImage: 'airplane',
        tintColor: markerColor,
        title: `${name} (${iataCode})`,
      },
    ],
    [name, latitude, longitude, iataCode, markerColor],
  )

  const cameraPosition = useMemo(
    () => ({
      coordinates: { latitude, longitude },
      zoom: zoomLevel,
    }),
    [latitude, longitude],
  )

  const localeStrings = useMemo(
    () => ({
      airportLocation: getLocale('airportLocation'),
    }),
    [selectedLocale],
  )

  const mapLocaleStrings = useMemo(
    () => ({
      appleMaps: getLocale('appleMaps'),
      cancel: getLocale('cancel'),
      googleMaps: getLocale('googleMaps'),
      selectMapApp: getLocale('selectMapApp'),
    }),
    [selectedLocale],
  )

  const { onMarkerClick } = useMapActions(mapLocaleStrings)

  const handleMarkerPress = useCallback(
    (marker: AppleMapsMarker) => {
      onMarkerClick(marker, name)
    },
    [onMarkerClick, name],
  )

  return (
    <View>
      <ThemedText className="mb-3" color="text-100" type="h3">
        {localeStrings.airportLocation}
      </ThemedText>

      <View className="rounded-xl overflow-hidden h-64">
        {Platform.OS === 'ios' ? (
          <AppleMaps.View
            cameraPosition={cameraPosition}
            markers={appleMarkers}
            onMarkerClick={handleMarkerPress}
            style={{ flex: 1 }}
          />
        ) : (
          <WebView
            key={id}
            originWhitelist={['*']}
            source={{ html: googleMapsHTML }}
            style={{ flex: 1 }}
            allowsInlineMediaPlayback
            domStorageEnabled
            javaScriptEnabled
            scalesPageToFit
            startInLoadingState
            useWebKit
          />
        )}
      </View>
    </View>
  )
}
