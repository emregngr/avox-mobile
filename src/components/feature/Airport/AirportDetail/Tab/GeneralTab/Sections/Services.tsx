import { MaterialCommunityIcons } from '@expo/vector-icons'
import React, { useMemo } from 'react'
import { View } from 'react-native'

import { ThemedText } from '@/components/common/ThemedText'
import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirportType } from '@/types/feature/airport'

interface ServicesProps {
  facilities: AirportType['facilities']
}

export const Services = ({ facilities }: ServicesProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { services } = facilities ?? {}

  if (!services?.length) return null

  const localeStrings = useMemo(
    () => ({
      services: getLocale('services'),
    }),
    [selectedLocale],
  )

  return (
    <AirportSectionRow title={localeStrings.services}>
      {services.map((service, index) => (
        <View className="flex-row py-1.5 gap-x-2" key={index}>
          <MaterialCommunityIcons
            color={colors?.onPrimary100}
            name="checkbox-marked-circle"
            size={16}
          />
          <ThemedText className="capitalize" color="text-90" type="body2">
            {service}
          </ThemedText>
        </View>
      ))}
    </AirportSectionRow>
  )
}
