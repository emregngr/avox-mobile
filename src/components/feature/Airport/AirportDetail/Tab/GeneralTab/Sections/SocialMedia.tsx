import * as Linking from 'expo-linking'
import React, { useCallback, useMemo } from 'react'
import { TouchableOpacity, View } from 'react-native'

import InstagramIcon from '@/assets/icons/instagram.svg'
import LinkedinIcon from '@/assets/icons/linkedin.svg'
import TiktokIcon from '@/assets/icons/tiktok.svg'
import XIcon from '@/assets/icons/x.svg'
import { AirportSectionRow } from '@/components/feature/Airport/AirportDetail/AirportSectionRow'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirportType } from '@/types/feature/airport'

interface SocialMediaProps {
  airportInfo: AirportType['airportInfo']
}

export const SocialMedia = ({ airportInfo }: SocialMediaProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { socialMedia: { instagram, linkedin, tiktok, x } = {} } = airportInfo ?? {}

  const handleSocialMediaPress = useCallback(async (url: string) => {
    await Linking?.openURL(url)
  }, [])

  const localeStrings = useMemo(
    () => ({
      socialMedia: getLocale('socialMedia'),
    }),
    [selectedLocale],
  )

  const socialMediaIcons = useMemo(
    () => (
      <View className="flex-row items-center justify-evenly mt-2">
        {instagram ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(instagram)}
            testID="Instagram"
          >
            <InstagramIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}

        {tiktok ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(tiktok)}
            testID="Tiktok"
          >
            <TiktokIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}

        {x ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(x)}
            testID="X"
          >
            <XIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}

        {linkedin ? (
          <TouchableOpacity
            activeOpacity={0.7}
            hitSlop={10}
            onPress={() => handleSocialMediaPress(linkedin)}
            testID="Linkedin"
          >
            <LinkedinIcon color={colors?.onPrimary100} height={36} width={36} />
          </TouchableOpacity>
        ) : null}
      </View>
    ),
    [colors?.onPrimary100, handleSocialMediaPress, instagram, tiktok, x, linkedin],
  )

  return <AirportSectionRow title={localeStrings.socialMedia}>{socialMediaIcons}</AirportSectionRow>
}
