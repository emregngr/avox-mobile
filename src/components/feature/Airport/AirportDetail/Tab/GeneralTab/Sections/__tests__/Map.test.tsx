import { render } from '@testing-library/react-native'
import React from 'react'
import { Platform } from 'react-native'

import { Map } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Map'
import { useMapActions } from '@/hooks/maps/useMapAction'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'

const mockedOnMarkerClick = jest.fn()

jest.mock('@/hooks/maps/useMapAction')

const mockedUseMapActions = useMapActions as jest.MockedFunction<typeof useMapActions>

jest.mock('@/store/locale')

const mockedUseLocaleStore = useLocaleStore as jest.MockedFunction<typeof useLocaleStore>

jest.mock('@/store/theme')

const mockedUseThemeStore = useThemeStore as jest.MockedFunction<typeof useThemeStore>

jest.mock('@/components/common/ThemedText', () => {
  const { Text } = require('react-native')

  return {
    ThemedText: ({ children, ...props }: { children: string }) => (
      <Text {...props}>{children}</Text>
    ),
  }
})

const mockedAirportData: any = {
  id: '123',
  iataCode: 'LTFM',
  name: 'Istanbul Airport',
  operations: {
    location: {
      coordinates: {
        latitude: 41.0082,
        longitude: 28.9784,
      },
    },
    region: 'europe',
  },
}

beforeEach(() => {
  mockedUseLocaleStore.mockReturnValue({
    selectedLocale: 'en',
  })
  mockedUseThemeStore.mockReturnValue({
    selectedTheme: 'light',
  })

  mockedUseMapActions.mockReturnValue({
    onMarkerClick: mockedOnMarkerClick,
  })
})

describe('Map Component', () => {
  describe('Platform: Android', () => {
    beforeAll(() => {
      Platform.OS = 'android'
    })

    it('renders WebView', () => {
      const { getByTestId } = render(<Map airportData={mockedAirportData} />)
      expect(getByTestId('mocked-webview')).toBeTruthy()
    })

    it('generates HTML with the correct, evaluated URL', () => {
      const { getByTestId } = render(<Map airportData={mockedAirportData} />)
      const webView = getByTestId('mocked-webview')
      const html = webView.props.source.html

      const expectedUrl = `src="https://maps.google.com/maps?q=${mockedAirportData.operations.location.coordinates.latitude},${mockedAirportData.operations.location.coordinates.longitude}&z=12&output=embed"`
      expect(html).toContain(expectedUrl)
    })
  })

  describe('Data Handling (Testing for Crashes)', () => {
    it('throws an error when AirportInfo is null', () => {
      expect(() => render(<Map airportData={null as any} />)).toThrow()
    })

    it('throws an error for missing operations', () => {
      const incompleteData: any = { id: '1', name: 'Test' }
      expect(() => render(<Map airportData={incompleteData} />)).toThrow(
        "Cannot read properties of undefined (reading 'coordinates')",
      )
    })

    it('throws an error for missing hub', () => {
      const incompleteData: any = { id: '1', name: 'Test', operations: {} }
      expect(() => render(<Map airportData={incompleteData} />)).toThrow(
        "Cannot read properties of undefined (reading 'coordinates')",
      )
    })

    it('throws an error for missing coordinates', () => {
      const incompleteData: any = {
        id: '1',
        name: 'Test',
        operations: { location: {} },
      }
      expect(() => render(<Map airportData={incompleteData} />)).toThrow(
        "Cannot read properties of undefined (reading 'latitude')",
      )
    })
  })
})

describe('Map Component Snapshot', () => {
  it('should render the Map Component successfully', () => {
    const { toJSON } = render(<Map airportData={mockedAirportData} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
