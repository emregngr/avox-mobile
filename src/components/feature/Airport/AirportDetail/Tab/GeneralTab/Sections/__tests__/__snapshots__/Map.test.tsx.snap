// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Map Component Snapshot should render the Map Component successfully 1`] = `
<View>
  <Text
    className="mb-3"
    color="text-100"
    type="h3"
  />
  <View
    className="rounded-xl overflow-hidden h-64"
  >
    <View
      allowsInlineMediaPlayback={true}
      domStorageEnabled={true}
      javaScriptEnabled={true}
      originWhitelist={
        [
          "*",
        ]
      }
      scalesPageToFit={true}
      source={
        {
          "html": "
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <title>Istanbul Airport HaritasÄ±</title>
      <style>
        body {
          margin: 0;
          padding: 0;
          font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
          background-color: #ffffff;
        }

        .map-container {
          width: 100%;
          height: 250px;
          position: relative;
          overflow: hidden;
          border: 1px solid #e0e0e0; border-radius: 8px;
        }

        iframe {
          width: 100%;
          height: 260px;
          border: none;
          position: absolute;
          top: 0;
          left: 0;
          
        }
      </style>
    </head>
    <body>
      <div class="map-container">
        <iframe
          src="https://maps.google.com/maps?q=41.0082,28.9784&z=12&output=embed"
          allowfullscreen=""
          loading="lazy"
          referrerpolicy="no-referrer-when-downgrade">
        </iframe>
      </div>
    </body>
    </html>
  ",
        }
      }
      startInLoadingState={true}
      style={
        {
          "flex": 1,
        }
      }
      testID="mocked-webview"
      useWebKit={true}
    />
  </View>
</View>
`;
